LibreTranslate.Net/LibreTranslate.cs

```diff
@@ -1,114 +1,73 @@
-        }
-
-        public Translate(string url)
+        /// <summary>
+        /// The http client
+        /// </summary>
+        private readonly HttpClient HttpClient;
+        /// <summary>
+        /// The default contructor. The default http client base uri points to https://libretranslate.com
+        /// </summary>
+        public LibreTranslate()
         {
-            wc = new WebClient();
-            LanguageList = new List<Language>();
-            Url = url;
-            var languages = wc.DownloadString($"{url}/languages");
-            if (languages.Contains("\"en\""))
-                LanguageList.Add(Language.En);
-            if (languages.Contains("\"ar\""))
-                LanguageList.Add(Language.Ar);
-            if (languages.Contains("\"zh\""))
-                LanguageList.Add(Language.Zh);
-            if (languages.Contains("\"fr\""))
-                LanguageList.Add(Language.Fr);
-            if (languages.Contains("\"de\""))
-                LanguageList.Add(Language.De);
-            if (languages.Contains("\"it\""))
-                LanguageList.Add(Language.It);
-            if (languages.Contains("\"pt\""))
-                LanguageList.Add(Language.Pt);
-            if (languages.Contains("\"ru\""))
-                LanguageList.Add(Language.Ru);
-            if (languages.Contains("\"es\""))
-                LanguageList.Add(Language.Es);
+            HttpClient = new HttpClient()
+            {
+                BaseAddress = new Uri("https://libretranslate.com"),
+                Timeout = System.Threading.Timeout.InfiniteTimeSpan
+            };
         }
-
-        private WebClient wc { get; }
-        private string Url { get; }
-        private List<Language> LanguageList { get; }
-
-        public string TranslateText(Language fromLang, Language toLang, string text)
+        /// <summary>
+        /// Contructor which enable to specified the libre translate server api address
+        /// </summary>
+        /// <param name="url"></param>
+        public LibreTranslate(string url)
         {
-            if (!LanguageList.Contains(fromLang) ||
-                !LanguageList.Contains(toLang)) //if server doesn't support either language
-                throw new Exception(
-                    $"Server doesn't support this language! {string.Join(',', LanguageList.ToArray())}");
-
-            var data =
-                $"q={Uri.EscapeDataString(text)}&source={fromLang.ToString().ToLower()}&target={toLang.ToString().ToLower()}";
-            wc.Headers.Add("Content-Type: application/x-www-form-urlencoded");
-            var response =
-                JsonSerializer.Deserialize<TranslationResponse>(wc.UploadString(Url + "/translate", data));
-            return response.translatedText;
+            HttpClient = new HttpClient()
+            {
+                BaseAddress = new Uri(url),
+                Timeout = System.Threading.Timeout.InfiniteTimeSpan
```

Comment:
Perhaps consider a more sane amount instead of infinite. This would result in a program halting forever without a warning of any kind, if the server is down, if the internet is disconnected, if there is a DNS issue, etc